lib/fpm/package/deb.rb

```diff
@@ -641,29 +641,39 @@ def write_scripts
   end # def write_scripts
 
   def write_conffiles
-    return unless config_files.any?
-
-    # scan all conf file paths for files and add them
     allconfigs = []
-    config_files.each do |path|
+
+    # expand recursively a given path to be put in allconfigs
+    def add_path(path, allconfigs)
       # Strip leading /
       path = path[1..-1] if path[0,1] == "/"
       cfg_path = File.expand_path(path, staging_path)
+      Find.find(cfg_path) do |p|
+        allconfigs << p.gsub("#{staging_path}/", '') if File.file? p
+      end
+    end
+
+    # scan all conf file paths for files and add them
+    config_files.each do |path|
       begin
-        Find.find(cfg_path) do |p|
-          allconfigs << p.gsub("#{staging_path}/", '') if File.file? p
-        end
-      rescue Errno::ENOENT => e
+        add_path path, allconfigs
+      rescue Errno::ENOENT
         raise FPM::InvalidPackageConfiguration,
           "Error trying to use '#{cfg_path}' as a config file in the package. Does it exist?"
       end
     end
+
+    # Also add everything in /etc
+    begin
+      add_path '/etc', allconfigs unless attributes[:no_config_files?]
```

Comment:
We should log that we are doing this, as you asked about the warning about the behavior change, maybe somethign like:

```ruby
# Written assuming '--no-config-files' stays as the flag name
if !attributes[:no_config_files?]
  logger.warn("Something about what we are doing, and why....")
  add_path("/etc", allconfigs)
end
  ```